i3lock.c

```diff
@@ -1743,23 +1806,28 @@ int main(int argc, char *argv[]) {
                                  (uint32_t[]){XCB_EVENT_MASK_STRUCTURE_NOTIFY});
 
     init_colors_once();
-    if (verify_png_image(image_path)) {
-        /* Create a pixmap to render on, fill it with the background color */
-        img = cairo_image_surface_create_from_png(image_path);
-    } else if (file_is_jpg(image_path)) {
-        DEBUG("Image looks like a jpeg, decoding\n");
-        unsigned char* jpg_data = read_JPEG_file(image_path, &jpg_info);
-            if (jpg_data != NULL) {
-                img = cairo_image_surface_create_for_data(jpg_data,
-                        CAIRO_FORMAT_ARGB32, jpg_info.width, jpg_info.height,
-                        jpg_info.stride);
-            }
-    }
-    /* In case loading failed, we just pretend no -i was specified. */
-    if (img && cairo_surface_status(img) != CAIRO_STATUS_SUCCESS) {
-        fprintf(stderr, "Could not load image \"%s\": %s\n",
-                image_path, cairo_status_to_string(cairo_surface_status(img)));
-        img = NULL;
+    if (is_regular_file(image_path)) {
+        if (verify_png_image(image_path)) {
```

Comment:
this code to load a single image into a buffer will get reused above in the load_slideshow thing (which didn't handle jpegs).

it's probably best to just dump this into a helper function:

```
cairo_surface_t* load_image(char* path) {
// copy code, return `NULL` on fail
}
```